src/consumer/runner.js

```diff
@@ -344,27 +340,19 @@ module.exports = class Runner {
 
               await onBatch(batch)
               await this.consumerGroup.heartbeat({ interval: this.heartbeatInterval })
-            } catch (e) {
-              unlockWithError(e)
-            } finally {
-              numberOfExecutions++
-              if (numberOfExecutions === expectedNumberOfExecutions) {
-                unlock()
-              }
-            }
-          }).catch(unlockWithError)
-        )
+            })
+          )
+        }
       })
     }
 
+    const enqueuedTasks = await Promise.all(allResults.map(fn => fn()))
+
     if (!this.running) {
       return
     }
 
-    await Promise.all(enqueuedTasks.map(fn => fn()))
-    if (expectedNumberOfExecutions > 0) {
-      await lock
-    }
+    await Promise.all([].concat(...enqueuedTasks.map(fn => fn())))
```

Comment:
Aren't you awaiting for this already at line 349?

Reply:
In line 349 we are creating an array `enqueuedTasks` we need to await for allResults array to make sure we fetch all `const batches = await result.value` without errors in case if it fails it will throw up the chain. In this line we are actually awaiting every single batch process